packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FGMGroundOverlayController.m

```diff
@@ -0,0 +1,240 @@
+  }
+  return self;
+}
+
+- (void)removeGroundOverlay {
+  self.groundOverlay.map = nil;
+}
+
+- (void)setConsumeTapEvents:(BOOL)consumes {
+  self.groundOverlay.tappable = consumes;
+}
+
+- (void)setVisible:(BOOL)visible {
+  self.groundOverlay.map = visible ? self.mapView : nil;
+}
+
+- (void)setZIndex:(int)zIndex {
+  self.groundOverlay.zIndex = zIndex;
+}
+
+- (void)setAnchor:(CGPoint)anchor {
+  self.groundOverlay.anchor = anchor;
+}
+
+- (void)setBearing:(CLLocationDirection)bearing {
+  self.groundOverlay.bearing = bearing;
+}
+
+- (void)setTransparency:(float)transparency {
+  float opacity = 1.0 - transparency;
+  self.groundOverlay.opacity = opacity;
+}
+
+- (void)setPositionFromBounds:(GMSCoordinateBounds *)bounds {
+  self.groundOverlay.bounds = bounds;
+}
+
+- (void)setPositionFromCoordinates:(CLLocationCoordinate2D)coordinates {
+  self.groundOverlay.position = coordinates;
+}
+
+- (void)setIcon:(UIImage *)icon {
+  self.groundOverlay.icon = icon;
+}
+
+- (void)updateFromPlatformGroundOverlay:(FGMPlatformGroundOverlay *)groundOverlay
+                              registrar:(NSObject<FlutterPluginRegistrar> *)registrar
+                            screenScale:(CGFloat)screenScale {
+  [self setConsumeTapEvents:groundOverlay.clickable];
+  [self setVisible:groundOverlay.visible];
+  [self setZIndex:(int)groundOverlay.zIndex];
+  [self setAnchor:CGPointMake(groundOverlay.anchor.x, groundOverlay.anchor.y)];
+  UIImage *image = [FGMUtils iconFromBitmap:groundOverlay.image
+                                  registrar:registrar
+                                screenScale:screenScale];
+  [self setIcon:image];
+  [self setBearing:groundOverlay.bearing];
+  [self setTransparency:groundOverlay.transparency];
+  if (_isCreatedWithBounds) {
+    [self setPositionFromBounds:[[GMSCoordinateBounds alloc]
+                                    initWithCoordinate:CLLocationCoordinate2DMake(
+                                                           groundOverlay.bounds.northeast.latitude,
+                                                           groundOverlay.bounds.northeast.longitude)
+                                            coordinate:CLLocationCoordinate2DMake(
+                                                           groundOverlay.bounds.southwest.latitude,
+                                                           groundOverlay.bounds.southwest
+                                                               .longitude)]];
+  } else {
+    [self setPositionFromCoordinates:CLLocationCoordinate2DMake(groundOverlay.position.latitude,
+                                                                groundOverlay.position.longitude)];
+  }
+}
+
+@end
+
+@interface FLTGroundOverlaysController ()
+
+/// A map from ground overlay id to the controller that manages it.
+@property(strong, nonatomic) NSMutableDictionary<NSString *, FGMGroundOverlayController *>
+    *groundOverlayIdentifierToController;
```

Comment:
Nit: if you name it `groundOverlayControllerByIdentifier`, then code doing lookups into it will read more like a method, where the last thing names the argument (or in this case, the index).

Reply:
Changed the name.